test/spectests/spectests.cpp

```diff
@@ -402,6 +396,20 @@ class test_runner
         return it_instance->second.get();
     }
 
+    std::optional<fizzy::Value> read_value(const json& v)
+    {
+        const auto arg_type = v.at("type").get<std::string>();
+        if (arg_type != "i32" && arg_type != "i64" && arg_type != "f32" && arg_type != "f64")
+        {
+            skip("Unsupported value type '" + arg_type + "'.");
+            return std::nullopt;
+        }
+
+        // Values of all types are serialized to JSON as integers.
+        // Value type will handle correct conversions.
```

Comment:
How? Why not keep the old `json_to_value<int32_t>(arg.at("value"))` way?

Reply:
It's just simpler without additional `switch` and template.

Presumably `wast2json` does the opposite conversion - typed value into unsigned integer. We could reinterpret_cast it back to correct type, but `Value(correct_type_value)` constructor does the same as `Value(reinterpret_cast<unsigned long long>(correct_type_value))`, which is the idea of `Value` type...